### PR TITLE
New package: alpgen

### PR DESCRIPTION
**Pull request by iarspider**

### CODE REVIEW

File: var/spack/repos/builtin/packages/alpgen/package.py [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+
+from spack import *
+
+
+class Alpgen(MakefilePackage):
+    """A collection of codes for the generation of
+       multi-parton processes in hadronic collisions."""
+
+    homepage = "http://mlm.home.cern.ch/mlm/alpgen/"
+    url      = "https://mlm.home.cern.ch/mlm/alpgen/V2.1/v214.tgz"
```
Comment (bernhardkaindl) [inline var/spack/repos/builtin/packages/alpgen/package.py]: Maybe you initially built the package with `http://` in `url = "https://`, or the cert just changed now:
```py
wget "https://mlm.home.cern.ch/mlm/alpgen/V2.1/v214.tgz"
--2021-10-14 13:20:36--  https://mlm.home.cern.ch/mlm/alpgen/V2.1/v214.tgz
Resolving mlm.home.cern.ch (mlm.home.cern.ch)... 2001:1458:d00:41::100:151, 188.185.90.198
Connecting to mlm.home.cern.ch (mlm.home.cern.ch)|2001:1458:d00:41::100:151|:443... connected.
ERROR: no certificate subject alternative name matches
        requested host name ‘mlm.home.cern.ch’.
To connect to mlm.home.cern.ch insecurely, use `--no-check-certificate'.
```
Depending on the python version (or some other difference) on the build host, the spack download fails just with an unexplained fetch failure or with
```py
==> Error: CertificateError: hostname 'mlm.home.cern.ch' doesn't match either of 'www.cern.ch', '*.app.cern.ch', '*.apptest.cern.ch', '*.docs.cern.ch', '*.web.cern.ch', 'about.cern', 'accelerators.cern', 'ams02.space', 'antimatter.cern', 'arts.cern', 'at.cern', 'atlas.cern', 'beamlineforschools.cern', 'careers.cern', 'cern.ch', 'cernandsocietyfoundation.cern', 'cernvm.cern', 'cixp.net', 'clear.cern', 'clic.cern', 'cms.cern', 'computing.cern', 'cosmicrays.cern', 'darkmatter.cern', 'dotcernpilot.info', 'education.cern', 'engineering.cern', 'espace.cern.ch', 'espace2013.cern.ch', 'europeanstrategy.cern', 'experiments.cern', 'exsp2owa.cern.ch', 'fluka.cern', 'giving.cern', 'go.cern', 'groups.cern.ch', 'higgsboson.cern', 'home.cern', 'hse.cern', 'ideasquare.cern', 'ippog.org', 'jobs.cern', 'knowledge.cern', 'kt.cern', 'learn.cern', 'lhc.cern', 'library.cern', 'linearcollider.org', 'news.cern', 'newsroom.cern', 'nic.cern', 'opendata.atlas.cern', 'opendays.cern', 'openlab.cern', 'openshift-dev.cern.ch', 'openshift.cern.ch', 'particles.cern', 'pensionfund.cern.ch', 'physics.cern', 'press.cern', 'pwa.cern.ch', 'science.cern', 'sciencegateway.cern', 'search.cern.ch', 'social.cern.ch', 'standardmodel.cern', 'supersymmetry.cern', 'teachers.cern', 'technology.cern', 'test-home.cern', 'united-states.cern', 'visit.cern', 'voisins.cern', 'www.arts.cern', 'www.atlas.cern', 'www.cern', 'www.home.cern', 'www.test-home.cern'
```
Therefore commit this suggestion with the button below and I would be happy:
```suggestion
    url      = "http://mlm.home.cern.ch/mlm/alpgen/V2.1/v214.tgz"
```

**Review by bernhardkaindl — CHANGES_REQUESTED** [review]

**Review by bernhardkaindl — APPROVED** [review]

**Comment by alalazo** [comment]
Is there any reason why, in this recipe, the package is a `MakefilePackage` but it builds either with a custom script or CMake? Why wasn't this coded as a `CMakePackage`?

**Comment by iarspider** [comment]
@alalazo to avoid dependency on `cmake` when using `recipe=cms`

**Comment by alalazo** [comment]
Why not a `Package` then? Asking since I'm looking into implementing support for multiple build-systems at once and currently trying to look into relevant use cases.

**Comment by iarspider** [comment]
@alalazo no special reason.

**Comment by iarspider** [comment]
> Asking since I'm looking into implementing support for multiple build-systems

Ah, I knew this! Good luck!

**Pull request merged**